src/imputation/tmi_imputation.py

```diff
@@ -100,51 +71,41 @@ def create_imp_class_col(
     return df
 
 
-def fill_zeros(df: pd.DataFrame, column: str) -> pd.DataFrame:
-    """Fills null values with zeros in a given column."""
-    return df[column].fillna(0).astype("float")
-
-
-def apply_fill_zeros(clear_df, df, target_variables: list):
+def apply_fill_zeros(df: pd.DataFrame, target_variables: list):
     """Applies the fill zeros function to filtered dataframes.
 
-    This will be applied to all target variables for clear responders
-    and all target variables for "No R&D".
+    A mask is created to identify clear responders, excluding instance zero rows, 
+    but exclude "postcode only" rows.
 
-    Args:
-        clear_df (pd.DataFrame): The clear responders, excluding instance 0.
-        df (pd.DataFrame): The unfiltered dataframe
-    """
-    # only fill zeros where intance is not zero
-    clear_df = clear_df.loc[clear_df["instance"] != 0]
+    Zeros are then filled for the target values based on this mask, and the equivalent 
+    "_imputed" columns.
 
-    no_rd = filter_by_column_content(df, "604", ["No"])
-    no_rd = no_rd.loc[no_rd["instance"] != 0]
+    Args:
+        df (pd.DataFrame): The dataframe imputation is carried out on
+        target_variables (List): A list of the target variables
 
-    for i in target_variables:
-        clear_df[i] = fill_zeros(clear_df, i)
-        clear_df[f"{i}_imputed"] = fill_zeros(clear_df, f"{i}_imputed")
-        no_rd[i] = fill_zeros(clear_df, i)
-        no_rd[f"{i}_imputed"] = fill_zeros(no_rd, f"{i}_imputed")
+    Returns:
+        pd.DataFrame: The same dataframe with required nulls filled with zeros.
+    """
+    zerofill_mask = (
+        (df["instance"] != 0) &
+        (df["status"].isin(["Clear", "Clear - overridden"])) &
+        (df["211"].isnull() & ~df["601"].isnull())
+    )
```

Comment:
This code is the main part of the ticket: The function fills nulls for clear responders, but the change is that  we shouldn't fill nulls for the "postcode only" rows.